lib/generator/models.ex

```diff
@@ -0,0 +1,172 @@
+      {full_name, generate_enum(full_name, enum)}
+    end
+  end
+
+  defp generate_enum(name, enum) do
+    parts = Enum.map(enum.values, fn {key, value} ->
+      macro_name = key |> to_string |> String.downcase |> String.to_atom
+      quote do
+        defmacro unquote(Macro.var(macro_name, nil)), do: unquote(value)
+      end
+    end)
+    quote do
+      defmodule unquote(name) do
+        @moduledoc unquote("Auto-generated Thrift enum #{enum.name}")
+        unquote_splicing(parts)
+      end
+    end
+  end
+
+  defp generate_structs(schema) do
+    for {name, struct} <- schema.structs do
+      full_name = namespace(schema) |> Module.concat(name)
+      {full_name, generate_struct(schema, full_name, struct)}
+    end
+  end
+
+  defp generate_struct(schema, name, struct) do
+    struct_parts = Enum.map(struct.fields, fn
+      %{name: name, default: nil, type: type} ->
+        {name, zero(schema, type)}
+      %{name: name, default: default} when not is_nil(default) ->
+        {name, default}
+    end)
+    binary_serializer = Binary.generate_serializer(name, schema.file_group, struct)
+    quote do
+      defmodule unquote(name) do
+        _ = unquote "Auto-generated Thrift struct #{struct.name}"
+        unquote_splicing(for field <- struct.fields do
+          quote do
+            _ = unquote "#{field.name} #{inspect field.type}"
+          end
+        end)
+        defstruct unquote(struct_parts)
+        def new, do: %__MODULE__{}
+        unquote(binary_serializer)
+      end
+    end
+  end
+
+  defp generate_exceptions(schema) do
+    for {name, struct} <- schema.exceptions do
+      full_name = namespace(schema) |> Module.concat(name)
+      {full_name, generate_exception(schema, full_name, struct)}
+    end
+  end
+
+  defp generate_exception(schema, name, struct) do
+    struct_parts = Enum.map(struct.fields, fn
+      %{name: name, default: nil, type: type} ->
+        {name, zero(schema, type)}
+      %{name: name, default: default} when not is_nil(default) ->
+        {name, default}
+    end)
+    binary_serializer = Binary.generate_serializer(name, schema.file_group, struct)
+    quote do
+      defmodule unquote(name) do
+        _ = unquote "Auto-generated Thrift exception #{struct.name}"
+        unquote_splicing(for field <- struct.fields do
+          quote do
+            _ = unquote "#{field.name} #{inspect field.type}"
+          end
+        end)
+        defstruct unquote(struct_parts)
+        def new, do: %__MODULE__{}
+        unquote(binary_serializer)
+      end
+    end
+  end
+
+  # Zero values for built-in types
+  defp zero(_schema, :bool), do: false
+  defp zero(_schema, :byte), do: 0
+  defp zero(_schema, :i8), do: 0
+  defp zero(_schema, :i16), do: 0
+  defp zero(_schema, :i32), do: 0
+  defp zero(_schema, :i64), do: 0
+  defp zero(_schema, :double), do: 0.0
+  defp zero(_schema, :string), do: ""
+  defp zero(_schema, :binary), do: ""
+  defp zero(_schema, :slist), do: ""
```

Comment:
Are we going to support `slist`? If we do, we should also include `senum`. I'm fine leaving them both out because they've been deprecated for a long time: https://thrift.apache.org/docs/idl#senum